feat: add support for DeleteExporter request handling in broker (#35324)

giampa91 opened the pull request:
## Description

- Implemented broker-side request handling for deleteExporter API
- Updated PartitionConfigurationManager and ZeebePartition to support exporter deletion

<!-- Describe the goal and purpose of this PR. -->

## Checklist

## Related issues

closes #35324

giampa91 committed:
feat: add support for DeleteExporter request handling in broker (#35324)

- Implemented broker-side request handling for deleteExporter API
- Updated PartitionConfigurationManager and ZeebePartition to support exporter deletion

closes #35324

giampa91 commented:
Hi @deepthidevaki,

I noticed that I had to duplicate some logic between the disableExporter and deleteExporter paths. 
If the current logic looks correct to you, I’d be happy to refactor it in a follow-up to make the implementation more DRY and reduce redundancy.

deepthidevaki reviewed (changes requested):
Great :rocket: Some smaller suggestions
- :x:  We should also handle concurrent deletion of exporters and transition of ExporterDirection. It looks like the existing code already covers this case. However, to ensure please add a test in ExporterDirectorPartitionTransisitionStepTest for concurrently deleted exporters. You can refer `shouldDisableExporterIfConfigChangedConcurrently` for inspiration.
- :wrench: You can simplify most of the the assert statements as I suggested. For future ~you can follow [this suggestion](https://camunda.slack.com/archives/C08F5RD5X8B/p1753348049719029?thread_ts=1753341596.085299&cid=C08F5RD5X8B) shared in the slack.~   you can make use of [this plugin](https://plugins.jetbrains.com/plugin/12195-concise-assertj-optimizing-nitpicker-cajon-), if you are an intellij user.

giampa91 committed:
fix: adapt ExporterDirectorPartitionTransitionStep

update method to deleteOrDisableOrEnableExportersIfConfigChanged
in order to handle delete and not just disable or enable.
Test updated

deepthidevaki commented on zeebe/broker/src/main/java/io/camunda/zeebe/broker/exporter/stream/ExporterDirector.java:
```diff
@@ -225,10 +225,24 @@ public ActorFuture<Void> resumeExporting() {
+   */
+  public ActorFuture<Void> deleteExporter(final String exporterId) {
+    return callRemoveExporter(exporterId);
+  }
```
We can just have one public method "removeExporter" in ExporterDirector that is used for disabling and deleting.

deepthidevaki commented on zeebe/broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterDeleteTest.java:
```diff
@@ -0,0 +1,125 @@
+import org.junit.Rule;
+import org.junit.Test;
+
+public class ExporterDeleteTest {
```
When we have just one method "removeExporter", there is no need for this test because it will be a duplicate of `ExporterDisableTest`.

giampa91 committed:
refactor: use removeExporter in ExporterDirector

deepthidevaki commented:
@giampa91 Would you have time to finish up this PR? If not I'm happy to take over and get it merged. 

I would like to complete this feature by end of this week because I want to include this feature in the upcoming alpha release scheduled for next week. For the feature, we have to merge this PR and work on the one remaining issue to expose the endpoints. So I would like to merge this by EOD tomorrow at the latest. Please let me know if you have time to answer my question above and make changes if necessary. If not, I will take over and merge your PR with minor changes if necessary.

Thanks a lot for your contribution. :rocket:
